sympy/solvers/inequalities.py

```diff
@@ -481,11 +482,24 @@ def valid(x):
         if valid(pt):
             sol_sets.append(Interval(start, end, True, True))
 
+        elif solns == []: # if no solutions for equation e
+            nsol_sets = _real_conditions(expr,gen)
+        cv = Intersection(*nsol_sets).subs(gen,_gen)
         rv = Union(*sol_sets).subs(gen, _gen)
-
+        rv = Union(rv, cv)
     return rv if not relational else rv.as_relational(_gen)
 
 
+def _real_conditions(expr, gen):
+    n_sol_sets = []
+    e = expr.gts - expr.lts
+    if e.is_Add:
+        for subexpr in e.args:
+            n_sol_sets.append(solve_univariate_inequality(subexpr > 0, gen, relational=False))
```

Comment:
since `e = expr.gts - expr.lts` , it makes sure that `subexpr > 0` since `expr.gts` give the equation on the `greater than side` : [reference](https://github.com/sympy/sympy/blob/master/sympy/core/relational.py#L451)